src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -182,10 +182,593 @@ private with sharing class RD2_ScheduleService_TEST {
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2018, 3, 17))
+                .build();
+
+        npe03__Recurring_Donation__c rd2 = rd.clone();
+        rd2.npe03__Recurring_Donation_Campaign__c = UTIL_UnitTestData_TEST.mockId(Campaign.getSObjectType());
+
+        RD2_ScheduleService schedule = new RD2_ScheduleService();
+        Boolean updateNeeded = schedule.recurringDonationScheduleUpdateNeeded(rd2, rd);
+
+        System.assertEquals(true, updateNeeded, 'recurringDonationScheduleUpdateNeeded should return true.');
+
+    }
+
+    /***
+    * @description Verifies update is NOT needed when contact on recurring donation is changed.
+    */
+    @isTest
+    private static void shouldDetermineNoUpdateNeededWhenContactChanged() {
+
+        npe03__Recurring_Donation__c rd =
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withMockId()
+                .withContact(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
+                .withAmount(400)
+                .withPaymentMethod('Check')
+                .withInstallmentPeriodWeekly()
+                .withInstallmentFrequency(1)
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2018, 3, 17))
+                .build();
+
+        npe03__Recurring_Donation__c rd2 = rd.clone();
+        rd2.npe03__Contact__c = UTIL_UnitTestData_TEST.mockId(Contact.getSObjectType());
+
+        RD2_ScheduleService schedule = new RD2_ScheduleService();
+        Boolean updateNeeded = schedule.recurringDonationScheduleUpdateNeeded(rd2, rd);
+
+        System.assertEquals(false, updateNeeded, 'recurringDonationScheduleUpdateNeeded should return false.');
+
+    }
+
+    /***
+    * @description Verifies update is NOT needed when status on updated recurring donation is closed.
+    */
+    @isTest
+    private static void shouldDetermineNoUpdateNeededWhenStatusClosed() {
+
+        npe03__Recurring_Donation__c rd =
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withMockId()
+                .withContact(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
+                .withAmount(400)
+                .withPaymentMethod('Check')
+                .withInstallmentPeriodWeekly()
+                .withInstallmentFrequency(1)
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2018, 3, 17))
+                .build();
+
+        npe03__Recurring_Donation__c rd2 = rd.clone();
+        rd2.Status__c = RD2_Constants.STATUS_CLOSED;
+
+        RD2_ScheduleService schedule = new RD2_ScheduleService();
+        Boolean updateNeeded = schedule.recurringDonationScheduleUpdateNeeded(rd2, rd);
+
+        System.assertEquals(false, updateNeeded, 'recurringDonationScheduleUpdateNeeded should return false.');
+
+    }
+
+
```

Comment:
If it's possible (given the really tight timeline), it would be great to have unit tests against the `getUpdatedSchedules` method. Maybe something like
- shouldCreateNewMonthlySchedule (when schedule changed)
- shouldInactivateCurrentSchedule (when the RD is changed to Closed)
- shouldChangeFromFirstAndFifteenthToWeekly

Reply:
I can complete these this evening.